sw/device/silicon_creator/lib/otbn_util.h

```diff
@@ -113,6 +104,54 @@ typedef struct otbn {
   uint32_t error_bits;
 } otbn_t;
 
+/**
+ * Generate the prefix to add to an OTBN symbol name used on the Ibex side
+ *
+ * The result is a pointer to Ibex's rodata that should be used to initialise
+ * memory for that symbol.
+ */
+#define OTBN_SYMBOL_PTR(app_name, sym) _otbn_local_app_##app_name##_##sym
```

Comment:
Now that there's no more address-space conversion, I think Ibex-space symbols are only ever needed to load the app; once the data is copied to DMEM, the OTBN-space addresses will suffice. So I'm wondering if it's possible not to expose the Ibex-space symbols at all, so that it's impossible to confuse them? Maybe the `OTBN_SYMBOL_PTR` could just be `#undef`ed at the end of this header, or could be inlined into `OTBN_DECLARE_APP_SYMBOLS` and `OTBN_APP_T_INIT`. The latter would repeat the location prefix, but that way it wouldn't look like anyone can or should use the Ibex-space addresses.

Reply:
Hmm, you might be right. I was expecting to need them, but maybe not! I'll push up a version in a minute to do that.

Reply:
Ah! `#undef` doesn't work here because of the way the C preprocessor works:
```c
#define foo 1
#define bar foo + 1
#undef foo
bar  // => expands to foo, which is undefined
```
I've left some of the helper macros in because they avoid some ugly repetition and let us name what's going on. But I've dropped the unused `otbn_ptr_t` type and chopped down the docstrings for the macros that shouldn't be used outside of the header.